File: frontend/src/types/event.ts
import type Ticket from './ticket'

interface Place {
  country: { name: string }
  address: string
}

interface Sport {
  id: string
  name: string
}

interface StarterItem {
  starter_item: string
}

interface Article {
  title: string
  text: string
}

interface Document {
  file: string
}

interface SocialLink {
  link: string
}

interface EventBasic {
  id: string
  banner: string
  title: string
  place: Place
  tickets: Ticket[]
  participation_start: Date
  participation_end: Date
  sport_type: Sport
}

interface DatabaseEvent extends EventBasic {
  about: string
  starter_items: StarterItem[]
  articles: Article[]
  documents: Document[]
  registration_start: Date
  regestration_end: Date
  social_links: SocialLink[]
}

export { type DatabaseEvent, type EventBasic }

File: event/src/schemas/event.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class EventListFilterSchema(BaseModel):
    search: Optional[str] = None
    sport_type: Optional[uuid.UUID] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    is_avaliable: Optional[bool] = None


class GetEventList(BaseModel):
    id: uuid.UUID
    banner: Optional[str]
    title: str
    place: "PlaceSchema"
    tickets: list["TicketSchema"]
    participation_start: datetime
    participation_end: datetime
    sport_type: "SportTypeSchema"


class GetEvent(GetEventList):
    about: str
    starter_items: list["StarterItemsSchema"]
    articles: list["ArticleSchema"]
    documents: Optional[list["DocumentSchema"]]
    registration_start: datetime
    registration_end: datetime
    social_links: list["SocialLinkSchema"]
    sport_type: "SportTypeSchema"


class SocialLinkSchema(BaseModel):
    link: str


class SportTypeSchema(BaseModel):
    id: uuid.UUID
    name: str
    description: str


class SportTypeNameSchema(BaseModel):
    name: str


class DocumentSchema(BaseModel):
    file: str
    name: str


class PlaceSchema(BaseModel):
    country: "CountrySchema"
    address: str


class CountrySchema(BaseModel):
    name: str


class TargetGroupSchema(BaseModel):
    gender: str
    minimal_age: int


class TicketSchema(BaseModel):
    id: uuid.UUID
    title: str
    extra_title: str
    price: int
    max_places: int
    sport_type: "SportTypeNameSchema"


class StarterItemsSchema(BaseModel):
    starter_item: str


class ArticleSchema(BaseModel):
    title: str
    text: str
